checks/evaluation/pinned_dependencies.go

```diff
@@ -139,17 +133,22 @@ func PinningDependencies(name string, c *checker.CheckRequest,
 		return checker.CreateRuntimeErrorResult(name, err)
 	}
 
-	// Scores may be inconclusive.
-	actionScore = maxScore(0, actionScore)
-	dockerFromScore = maxScore(0, dockerFromScore)
-	dockerDownloadScore = maxScore(0, dockerDownloadScore)
-	scriptScore = maxScore(0, scriptScore)
-	pipScore = maxScore(0, pipScore)
-	npmScore = maxScore(0, npmScore)
-	goScore = maxScore(0, goScore)
+	// If no dependencies of an ecossystem are found, it results in an inconclusive score.
+	// We filter out inconclusive scores so only applicable ecossystems are considered in
+	// the final score.
+	scores := []int{actionScore, dockerFromScore, dockerDownloadScore, scriptScore, pipScore, npmScore, goScore}
```

Comment:
before this, it was common to forget ecosystems. what do you think about refactoring the logic in a way to avoid needing to remember to update this line in the future?

Reply:
Yes, I totally agree. I just think we would make lots of changes for pinned dependencies evaluation, since we would be needing to replace scoring for individual ecossystems by an iteration of all ecossystems and applying a score for them. So I was thinking of changing the evaluaiton in a second PR, if this one works out, but we can do it all at once if you think the changes need to come together.

Reply:
It seems like one change to me. If we're going to make major changes to how we score pinned dependencies, might as well do it in one change, not two.

Reply:
Ack, I'm working on these changes